packages/main/src/DateRangePicker.js

```diff
@@ -554,28 +534,28 @@ class DateRangePicker extends DatePicker {
 
 		// Collect both dates and merge them into one
 		if (this._firstDateTimestamp !== this._lastDateTimestamp || this._oneTimeStampSelected) {
-			this.value = this._formatValue(calStartDate.toLocalJSDate().valueOf() / 1000, calEndDate.toLocalJSDate().valueOf() / 1000);
+			this.value = this._formatValue(calStartDate.toLocalJSDate(), calEndDate.toLocalJSDate());
 		}
 
-		this.realValue = this._formatValue(calStartDate.toLocalJSDate().valueOf() / 1000, calEndDate.toLocalJSDate().valueOf() / 1000);
+		this.realValue = this._formatValue(calStartDate.toLocalJSDate(), calEndDate.toLocalJSDate());
 		this._prevValue = this.realValue;
 	}
 
 	/**
 	 * Combines the start and end dates of a range into a formated string
 	 *
-	 * @param {int} firstDateValue locale start date timestamp
-	 * @param {int} lastDateValue locale end date timestamp
+	 * @param {int} firstDate locale start date
+	 * @param {int} lastDate locale end date
 	 * @returns {string} formated start to end date range
 	 */
-	_formatValue(firstDateValue, lastDateValue) {
+	_formatValue(firstDate, lastDate) {
 		let value = "";
 		const delimiter = this.delimiter,
 			format = this.getFormat(),
-			firstDateString = format.format(new Date(firstDateValue * 1000)),
-			lastDateString = format.format(new Date(lastDateValue * 1000));
+			firstDateString = format.format(firstDate),
```

Comment:
Should we check if "firstDate" is present as we do for "lastDate" on the next line:
`firstDateString = firstDate && format.format(firstDate)`. Maybe it is always present? I think it is better to have the check to ensure the code won't throw.

And perhaps it would be better if the `if (firstDate)` becomes `if (firstDateString)` as at the end the logic in the statement is using `firstDateString` .

Reply:
Done

Reply:
In the component code, there isn't such case when firstDate argument isn't supplied. I agree that the described check needs to be reinforced for future uses of the method.